Crash for import azurerm_public_ip

_This issue was originally opened by @tvaroska as hashicorp/terraform#15898. It was migrated here as a result of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

terraform version 0.10.2

result of:
 terraform import azurerm_public_ip.console /subscriptions/$ARM_SUBSCRIPTION_ID/resourceGroups/0_central/providers/Microsoft.Network/publicIPAdresses/console-ip


panic: runtime error: invalid memory address or nil pointer dereference
2017-08-24T04:33:26.188Z [DEBUG] plugin.terraform-provider-azurerm_v0.1.5_x4: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xd113fb]
2017-08-24T04:33:26.188Z [DEBUG] plugin.terraform-provider-azurerm_v0.1.5_x4:
2017-08-24T04:33:26.188Z [DEBUG] plugin.terraform-provider-azurerm_v0.1.5_x4: goroutine 42 [running]:
2017-08-24T04:33:26.189Z [DEBUG] plugin.terraform-provider-azurerm_v0.1.5_x4: github.com/terraform-providers/terraform-provider-azurerm/azurerm.resourceArmPublicIpRead(0xc420112000, 0xec5860, 0xc42029e000, 0x0, 0x15b5bc0)
2017-08-24T04:33:26.189Z [DEBUG] plugin.terraform-provider-azurerm_v0.1.5_x4:   /opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/resource_arm_public_ip.go:171 +0x3eb
2017-08-24T04:33:26.189Z [DEBUG] plugin.terraform-provider-azurerm_v0.1.5_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc4201dde00, 0xc420052230, 0xec5860, 0xc42029e000, 0xc42025e648, 0xc420335e01, 0x0)
2017-08-24T04:33:26.189Z [DEBUG] plugin.terraform-provider-azurerm_v0.1.5_x4:   /opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:314 +0x21d
2017-08-24T04:33:26.188Z [DEBUG] plugin: plugin process exited: path=/home/boris/infrastructure/.terraform/plugins/linux_amd64/terraform-provider-azurerm_v0.1.5_x4
2017-08-24T04:33:26.189Z [DEBUG] plugin.terraform-provider-azurerm_v0.1.5_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc4201e07e0, 0xc4200521e0, 0xc420052230, 0x7f4128ebb000, 0x0, 0x0)
2017-08-24T04:33:26.189Z [DEBUG] plugin.terraform-provider-azurerm_v0.1.5_x4:   /opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:267 +0x91
2017-08-24T04:33:26.189Z [DEBUG] plugin.terraform-provider-azurerm_v0.1.5_x4: github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Refresh(0xc42010aa60, 0xc4203bfb30, 0xc4203bfdd0, 0x0, 0x0)
2017-08-24T04:33:26.189Z [DEBUG] plugin.terraform-provider-azurerm_v0.1.5_x4:   /opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:510 +0x4e
2017-08-24T04:33:26.189Z [DEBUG] plugin.terraform-provider-azurerm_v0.1.5_x4: reflect.Value.call(0xc42001fbc0, 0xc42000c2c8, 0x13, 0xf8147c, 0x4, 0xc420566f20, 0x3, 0x3, 0xfc28f0, 0xc4204c36e8, ...)
2017-08-24T04:33:26.189Z [DEBUG] plugin.terraform-provider-azurerm_v0.1.5_x4:   /usr/local/go/src/reflect/value.go:434 +0x91f
2017-08-24T04:33:26.189Z [DEBUG] plugin.terraform-provider-azurerm_v0.1.5_x4: reflect.Value.Call(0xc42001fbc0, 0xc42000c2c8, 0x13, 0xc4204c3720, 0x3, 0x3, 0xc4204c370c, 0x5355206c61727401, 0x200222c22)
2017-08-24T04:33:26.189Z [DEBUG] plugin.terraform-provider-azurerm_v0.1.5_x4:   /usr/local/go/src/reflect/value.go:302 +0xa4
2017-08-24T04:33:26.189Z [DEBUG] plugin.terraform-provider-azurerm_v0.1.5_x4: net/rpc.(*service).call(0xc42011b5c0, 0xc42011b580, 0xc4201da180, 0xc42006ff80, 0xc42010b500, 0xdfa760, 0xc4203bfb30, 0x16, 0xdfa7a0, 0xc4203bfdd0, ...)
2017-08-24T04:33:26.189Z [DEBUG] plugin.terraform-provider-azurerm_v0.1.5_x4:   /usr/local/go/src/net/rpc/server.go:387 +0x144
2017-08-24T04:33:26.189Z [DEBUG] plugin.terraform-provider-azurerm_v0.1.5_x4: created by net/rpc.(*Server).ServeCodec
2017-08-24T04:33:26.189Z [DEBUG] plugin.terraform-provider-azurerm_v0.1.5_x4:   /usr/local/go/src/net/rpc/server.go:481 +0x404

--- a/azurerm/import_arm_public_ip_test.go
+++ b/azurerm/import_arm_public_ip_test.go
@@ -1,6 +1,9 @@
 package azurerm
 
 import (
+	"fmt"
+	"os"
+	"regexp"
 	"testing"
 
 	"github.com/hashicorp/terraform/helper/acctest"
@@ -25,6 +28,30 @@ func TestAccAzureRMPublicIpStatic_importBasic(t *testing.T) {
 				ResourceName:      resourceName,
 				ImportState:       true,
 				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAzureRMPublicIpStatic_importIdError(t *testing.T) {
+	resourceName := "azurerm_public_ip.test"
+
+	ri := acctest.RandInt()
+	config := testAccAzureRMPublicIPStatic_basic(ri, testLocation())
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMPublicIpDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateId:     fmt.Sprintf("/subscriptions/%s/resourceGroups/acctestRG-%d/providers/Microsoft.Network/publicIPAdresses/acctestpublicip-%d", os.Getenv("ARM_SUBSCRIPTION_ID"), ri, ri),
+				ExpectError:       regexp.MustCompile("Error parsing supplied resource id."),
 			},
 		},
 	})

--- a/azurerm/resource_arm_public_ip.go
+++ b/azurerm/resource_arm_public_ip.go
@@ -18,7 +18,17 @@ func resourceArmPublicIp() *schema.Resource {
 		Update: resourceArmPublicIpCreate,
 		Delete: resourceArmPublicIpDelete,
 		Importer: &schema.ResourceImporter{
-			State: schema.ImportStatePassthrough,
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				id, err := parseAzureResourceID(d.Id())
+				if err != nil {
+					return nil, err
+				}
+				name := id.Path["publicIPAddresses"]
+				if name == "" {
+					return nil, fmt.Errorf("Error parsing supplied resource id. Please check it and rerun:\n %s", d.Id())
+				}
+				return []*schema.ResourceData{d}, nil
+			},
 		},
 
 		Schema: map[string]*schema.Schema{

